File: telegram-bot-bumblebee-core/src/main/java/com/github/bumblebee/command/currency/domain/SupportedCurrency.java
package com.github.bumblebee.command.currency.domain;

public enum SupportedCurrency {
    USD, EUR,

    AUD,
    BGN,
    UAH,
    DKK,
    PLN,
    IRR,
    ISK,
    JPY,
    CAD,
    CNY,
    KWD,
    MDL,
    NZD,
    NOK,
    RUB,
    XDR,
    SGD,
    KGS,
    KZT,
    TRY,
    GBP,
    CZK,
    SEK,
    CHF;

    public static SupportedCurrency parse(String currency) {
        if (currency == null) {
            return null;
        }

        for (SupportedCurrency supportedCurrency : SupportedCurrency.values()) {
            if (supportedCurrency.name().equalsIgnoreCase(currency)) {
                return supportedCurrency;
            }
        }
        return null;
    }
}

File: telegram-bot-bumblebee-core/src/main/kotlin/com/github/bumblebee/command/currency/domain/SupportedCurrency.kt
package com.github.bumblebee.command.currency.domain

enum class SupportedCurrency {
    USD,
    EUR,
    AUD,
    BGN,
    UAH,
    DKK,
    PLN,
    IRR,
    ISK,
    JPY,
    CAD,
    CNY,
    KWD,
    MDL,
    NZD,
    NOK,
    RUB,
    XDR,
    SGD,
    KGS,
    KZT,
    TRY,
    GBP,
    CZK,
    SEK,
    CHF;

    companion object {
        fun parse(currency: String?): SupportedCurrency? {
            return currency?.let {
                SupportedCurrency.values().firstOrNull { it.name.equals(currency, ignoreCase = true) }
            }
        }
    }
}
